public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -27,66 +26,66 @@ const SubmissionsTable = () => {
-                        <td className={SubmissionStatus[status]}>
-                            {assignmentStatusText(assignment, submission, status)}
-                        </td>
-                    </tr>
-                )
-            }
-        })
-        return table
-    }
+    const table: React.JSX.Element[] = []
+    sortedAssignments().forEach(assignment => {
+        if (isExpired(assignment)) {
+            return // ignore expired assignments
+        }
+        const courseID = assignment.CourseID
+        const submissions = state.submissions.ForAssignment(assignment)
+        if (submissions.length === 0) {
+            return
+        }
+        if (state.enrollmentsByCourseID[courseID.toString()]?.status !== Enrollment_UserStatus.STUDENT) {
+            return
+        }
+        // Submissions are indexed by the assignment order - 1.
+        const submission = submissions.find(sub => sub.AssignmentID === assignment.ID) ?? create(SubmissionSchema)
+        const status = getStatusByUser(submission, state.self.ID)
+        if (!isApproved(status)) {
+            const deadline = deadlineFormatter(assignment, submission.score)
+            const course = state.courses.find(c => c.ID === courseID)
+            table.push(
+                <tr key={assignment.ID.toString()} className={`clickable-row ${deadline.className}`}
+                    onClick={() => history.push(`/course/${courseID}/lab/${assignment.ID}`)}>
+                    <th scope="row">{course?.code}</th>
+                    <td>
+                        {assignment.name}
+                        {assignment.isGroupLab ?
+                            <span className="badge ml-2 float-right"><i className={Icon.GROUP} title="Group Assignment" /></span> : null}
+                    </td>
+                    <td><ProgressBar courseID={courseID.toString()} submission={submission} type={Progress.OVERVIEW} /></td>
+                    <td>{deadline.time}</td>
+                    <td>{deadline.message}</td>
+                    <td className={SubmissionStatus[status]}>
+                        {assignmentStatusText(assignment, submission, status)}
+                    </td>
+                </tr>
+            )
+        }
+    })
 
     return (
-        <div>
-            <table className="table rounded-lg table-bordered table-hover" id="LandingPageTable">
-                <thead >
-                    <tr>
-                        <th scope="col">Course</th>
-                        <th scope="col">Assignment</th>
-                        <th scope="col">Progress</th>
-                        <th scope="col">Deadline</th>
-                        <th scope="col">Due in</th>
-                        <th scope="col">Status</th>
-                    </tr>
-                </thead>
-                <tbody>
-                    {NewSubmissionsTable()}
-                </tbody>
-            </table>
-        </div>
+        table.length !== 0 ? (
```

Comment:
This ternary check is unnecessary. Prefer the guard style when there is a chance that something should return null.
```tax
if (table.length === 0) {
    return null
}
return (
     <div>
...
```